hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/OptimizerContext.java

```diff
@@ -103,34 +104,33 @@ private OptimizerContext(
      * Create the optimization context.
      *
      * @param searchPaths Search paths to support "current schema" feature.
-     * @param memberCount Number of member that is important for distribution-related rules and converters.
      * @return Context.
      */
     public static OptimizerContext create(
             SqlCatalog schema,
             List<List<String>> searchPaths,
             List<Object> arguments,
-            int memberCount,
-            IMapResolver iMapResolver
+            IMapResolver iMapResolver,
+            SqlSecurityContext securityContext
     ) {
         // Resolve tables.
         HazelcastSchema rootSchema = HazelcastSchemaUtils.createRootSchema(schema);
 
-        return create(rootSchema, searchPaths, arguments, memberCount, iMapResolver);
+        return create(rootSchema, searchPaths, arguments, iMapResolver, securityContext);
     }
 
     public static OptimizerContext create(
             HazelcastSchema rootSchema,
             List<List<String>> schemaPaths,
             List<Object> arguments,
-            int memberCount,
-            IMapResolver iMapResolver
+            IMapResolver iMapResolver,
+            SqlSecurityContext securityContext
     ) {
         Prepare.CatalogReader catalogReader = createCatalogReader(rootSchema, schemaPaths);
         HazelcastSqlValidator validator = new HazelcastSqlValidator(catalogReader, arguments, iMapResolver);
         VolcanoPlanner volcanoPlanner = createPlanner();
 
-        HazelcastRelOptCluster cluster = createCluster(volcanoPlanner);
+        HazelcastRelOptCluster cluster = createCluster(volcanoPlanner, securityContext);
```

Comment:
My intention is to keep it in `HazelcastRelOptCluster`, even if is not used atm.

Reply:
ok, but you still need to pass _current_ user to `execute` method, and you do not have `OptimizerContext` there